pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -1107,106 +1292,95 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListenerForPortOrUDS(l
-				if existingFilterChain.FilterChainMatch == nil {
-					// This is a catch all filter chain.
-					// We can only merge with a non-catch all filter chain
-					// Else mark it as conflict
-					if incomingFilterChain.FilterChainMatch == nil {
-						// NOTE: While pluginParams.Service can be nil,
-						// this code cannot be reached if Service is nil because a pluginParams.Service can be nil only
-						// for user defined Egress listeners with ports. And these should occur in the API before
-						// the wildcard egress listener. the check for the "locked" bit will eliminate the collision.
-						// User is also not allowed to add duplicate ports in the egress listener
-						var newHostname config.Hostname
-						if pluginParams.Service != nil {
-							newHostname = pluginParams.Service.Hostname
-						} else {
-							// user defined outbound listener via sidecar API
-							newHostname = "sidecar-config-egress-tcp-listener"
-						}
-
-						conflictFound = true
-						outboundListenerConflict{
-							metric:          model.ProxyStatusConflictOutboundListenerTCPOverTCP,
-							node:            pluginParams.Node,
-							listenerName:    listenerMapKey,
-							currentServices: currentListenerEntry.services,
-							currentProtocol: currentListenerEntry.servicePort.Protocol,
-							newHostname:     newHostname,
-							newProtocol:     pluginParams.Port.Protocol,
-						}.addMetric(pluginParams.Push)
-						break compareWithExisting
-					} else {
-						continue
-					}
-				}
-				if incomingFilterChain.FilterChainMatch == nil {
-					continue
-				}
-
-				// We have two non-catch all filter chains. Check for duplicates
-				if reflect.DeepEqual(*existingFilterChain.FilterChainMatch, *incomingFilterChain.FilterChainMatch) {
-					var newHostname config.Hostname
-					if pluginParams.Service != nil {
-						newHostname = pluginParams.Service.Hostname
-					} else {
-						// user defined outbound listener via sidecar API
-						newHostname = "sidecar-config-egress-tcp-listener"
-					}
-
-					conflictFound = true
-					outboundListenerConflict{
-						metric:          model.ProxyStatusConflictOutboundListenerTCPOverTCP,
-						node:            pluginParams.Node,
-						listenerName:    listenerMapKey,
-						currentServices: currentListenerEntry.services,
-						currentProtocol: currentListenerEntry.servicePort.Protocol,
-						newHostname:     newHostname,
-						newProtocol:     pluginParams.Port.Protocol,
-					}.addMetric(pluginParams.Push)
-					break compareWithExisting
-				}
-			}
-
-			if !conflictFound {
-				// There is no conflict with any filter chain in the existing listener.
-				// So append the new filter chains to the existing listener's filter chains
-				newFilterChains = append(newFilterChains, incomingFilterChain)
-				if pluginParams.Service != nil {
-					lEntry := listenerMap[listenerMapKey]
-					lEntry.services = append(lEntry.services, pluginParams.Service)
-				}
+	// There are 9 types conflicts
```

Comment:
this entire logic in the switch case: do you think it would be clearer if you used it at the point of conflict (in buildSidecarOutboundHTTPListenerForPortOrUDS or in TCPListener ? this will make it a bit more easier to follow. I leave it upto you. no strong preference.

Reply:
I was thinking to add the logic at the point of conflict but we still need code here to handle TCP over TCP, AUTO. I would prefer keep entire logic together.